pygmt/tests/test_session_management.py

```diff
@@ -20,3 +20,35 @@ def test_begin_end():
     begin()  # Restart the global session
     assert os.path.exists("pygmt-session.pdf")
     os.remove("pygmt-session.pdf")
+
+
+def test_gmt_compat_6_is_applied(capsys):
+    """
+    Ensure that users with old gmt.conf files won't get pygmt-session [ERROR]:
+    GMT_COMPATIBILITY: Expects values from 6 to 6; reset to 6.
+    """
+    end()  # Kill the global session
+    try:
+        with Session() as lib:
+            # pretend that gmt.conf has GMT_COMPATIBILITY = 5
+            lib.call_module("gmtset", "GMT_COMPATIBILITY 5")
+        begin()
+        with Session() as lib:
+            lib.call_module("basemap", "-R10/70/-3/8 -JX4i/3i -Ba")
+            out, err = capsys.readouterr()  # capture stdout and stderr
+            assert out == ""
+            assert err != (
+                "pygmt-session [ERROR]: GMT_COMPATIBILITY:"
+                " Expects values from 6 to 6; reset to 6.\n"
+            )
+            assert err == ""  # double check that there are no other errors
+    finally:
+        with Session() as lib:
+            # revert gmt.conf back to GMT_COMPATIBILITY = 6
+            lib.call_module("set", "GMT_COMPATIBILITY 6")
+        end()
+        begin()  # Restart the global session
+        assert os.path.exists("gmt.conf")
+        os.remove("gmt.conf")
```

Comment:
I don't think there is a "gmt.conf" file in the current directory. It should be in the temporary gmt session directory.

Reply:
I kinda agree, but wouldn't the assertion fail if there isn't a `gmt.conf` file? :confused: Things got really confusing for me when I was designing this test with `gmt.conf` files all over the place, one kept popping up in my `pygmt` folder with `GMT_COMPATIBILITY 5` for some reason.

Reply:
Ok, you're right, it's in "/home/.../pygmt/tmp-test-dir-with-unique-name/gmt.conf". I'll remove these lines once the tests on Windows pass in the other PR.

Reply:
I still don't understand why the test works, but since the test pass, OK to merge.

Reply:
Yeah, it's a complicated test. but removing the line at https://github.com/GenericMappingTools/pygmt/blob/8df13da7d7163ed265c22b7de18e1df344ec3664/pygmt/session_management.py#L19 should fail this test, as encountered by Mark.